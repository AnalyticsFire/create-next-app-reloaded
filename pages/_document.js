// ./pages/_document.js
import React from 'react';
import Document, { Head, Main, NextScript } from 'next/document';
import htmlescape from 'htmlescape';
import config from 'config';

export default class MyDocument extends Document {
  render() {
    return (
      <html>
        <Head>
          <link rel="stylesheet" href="/_next/static/style.css" />
        </Head>
        <body>
          <Main />
          <script
            // eslint-disable-next-line react/no-danger
            dangerouslySetInnerHTML={{
              __html: `
                __CONFIG__ = ${htmlescape(config)}
              `,
            }}
          />
          <NextScript />
        </body>
      </html>
    );
  }
}
