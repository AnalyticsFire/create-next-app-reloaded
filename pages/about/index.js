import Component, { About } from './about';

export { About };
export default Component;
