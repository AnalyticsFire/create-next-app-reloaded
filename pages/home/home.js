import Layout from '../../components/layout';

const Home = () => (
  <Layout title="Home" className="home-page">
    <h1>Home</h1>
  </Layout>
);

export default Home;
