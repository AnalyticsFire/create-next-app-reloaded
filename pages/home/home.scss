.home-page {
  h1 {
    font-size: $titleSize;
  }
}