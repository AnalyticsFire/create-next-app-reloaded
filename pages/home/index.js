import Component, { Home } from './home';

export { Home };
export default Component;
