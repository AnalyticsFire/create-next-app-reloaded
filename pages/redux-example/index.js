import Component, { ReduxExample } from './reduxExample';

export { ReduxExample };
export default Component;
