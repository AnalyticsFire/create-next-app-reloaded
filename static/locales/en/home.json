{
  "pageTitle": "AnalyticsFire - Home Page",
  "title":
    "This page was made using NextJs. It's currently running in {{ env }} mode!"
}
