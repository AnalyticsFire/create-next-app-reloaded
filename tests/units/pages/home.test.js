/* eslint-disable no-undef */

import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Home } from '@root/pages/home';

const defaultComponent = <Home t={() => {}} />;

test('Home is rendered', () => {
  expect(toJson(shallow(defaultComponent))).toMatchSnapshot();
});

/* eslint-enable no-undef */
