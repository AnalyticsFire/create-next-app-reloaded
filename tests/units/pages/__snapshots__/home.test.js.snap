// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home is rendered 1`] = `
<Layout
  className="home-page"
>
  <div
    className="container pt-4 pb-4"
  >
    <h1
      className="title text-center pb-3"
    />
    <img
      className="w-100"
      src="static/images/the-team.jpg"
    />
  </div>
</Layout>
`;
